packages/langium-cli/test/generator/ast-generator.test.ts

```diff
@@ -0,0 +1,109 @@
+/******************************************************************************
+ * Copyright 2023 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+import { describe, expect, test } from 'vitest';
+import { createLangiumGrammarServices, EmptyFileSystem, Grammar, normalizeEOL } from 'langium';
+import { parseHelper } from 'langium/test';
+import { LangiumConfig, RelativePath } from '../../src/package';
+import { generateAst } from '../../src/generator/ast-generator';
+
+const services = createLangiumGrammarServices(EmptyFileSystem);
+
+describe('Types generator', () => {
+
+    test('should generate checker functions for datatype rules', async () => {
+        const result = (await parseHelper<Grammar>(services.grammar)(TEST_GRAMMAR)).parseResult;
+        const config: LangiumConfig = {
+            [RelativePath]: './',
+            projectName: 'test',
+            languages: []
+        };
+        const typesFileContent = generateAst(services.grammar, [result.value], config);
+        expect(typesFileContent).toEqual(EXPECTED_AST_FILE);
+    });
+});
+
+const TEST_GRAMMAR = `
+grammar Arithmetics
+
+A returns string:
+    'a';
+
+AB returns string:
+    A | 'b';
+
+ABC returns string:
+    AB | ID;
+
+hidden terminal WS: /\\s+/;
+terminal ID: /[_a-zA-Z][\\w_]*/;
+
+hidden terminal ML_COMMENT: /\\/\\*[\\s\\S]*?\\*\\//;
+hidden terminal SL_COMMENT: /\\/\\/[^\\n\\r]*/;
+`;
+
+const EXPECTED_AST_FILE = normalizeEOL(`/******************************************************************************
```

Comment:
You can use `expandToString` from `template-string.ts` here to properly indent and format the expectation text, like
```
const EXPECTED_AST_FILE = expandToString`
    /*******
    ....
`
```

(not sure whether `normalizeEOL` would still be required for the test to pass on windows).